src/shogun/machine/IterativeLinearMachine.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+* This software is distributed under BSD 3-clause license (see LICENSE file).
+*
+* Authors: Shubham Shukla
+*/
+
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/common.h>
+#include <shogun/machine/IterativeLinearMachine.h>
+
+using namespace shogun;
+
+CIterativeLinearMachine::CIterativeLinearMachine() : CLinearMachine()
+{
+ m_current_iteration = 0;
+ m_complete = false;
+ register_params();
+}
+
+CIterativeLinearMachine::~CIterativeLinearMachine()
+{
+}
+
+bool CIterativeLinearMachine::train(CDotFeatures* f, CLabels* l)
+{
+   init_model(f, l);
+   continue_train(f, l);
+   return m_complete;
+}
+
+void CIterativeLinearMachine::continue_train(CDotFeatures* f, CLabels* l)
+{
+  //resume_computation();
+   while (m_current_iteration < m_max_iterations && !m_complete)
+   {
+       COMPUTATION_CONTROLLERS
+       iteration(f, l);
+       m_current_iteration++;
+   }
+}
+
+void CIterativeLinearMachine::register_params()
+{
+  SG_ADD(&m_current_iteration, "current_iteration", "current iteration", MS_AVAILABLE);
+  SG_ADD(&m_max_iterations, "max_iteration", "maximum iteration", MS_AVAILABLE);
```

Comment:
type in description.

model selection should only be available for max_iterations, not for the other two flags